rmf_traffic/src/rmf_traffic/Conflict.cpp

```diff
@@ -121,54 +121,116 @@ std::vector<ConflictData> DetectConflict::between(
 }
 
 //==============================================================================
-bool DetectConflict::broad_phase(
-    const Trajectory& trajectory_a,
-    const Trajectory& trajectory_b)
+
+namespace {
+
+using BoundingBox = std::pair<Eigen::Vector2d, Eigen::Vector2d>;
+
+double evaluate_spline(
+    const Eigen::Vector4d& coeffs,
+    const double t)
 {
-  std::size_t min_size = std::min(trajectory_a.size(), trajectory_b.size());
-  if(min_size < 2)
+  // Assume time is parameterized [0,1]
+  return (coeffs[3] * t * t * t
```

Comment:
That was exactly the misconception I had. But when I switch to the loop, interestingly I do get slightly slower performance as seen in the screenshots below (maybe not significant). I'm building with `--cmake-args -DCMAKE_BUILD_TYPE=RELEASE`.
Current:
![image](https://user-images.githubusercontent.com/13482049/72949215-97537400-3dc2-11ea-983d-e091ea5cad4d.png)

With loop:
![image](https://user-images.githubusercontent.com/13482049/72949238-afc38e80-3dc2-11ea-84bc-3604f1eb8b39.png)

Reply:
Ah wow. After some more reading, it seems that it may be the case that `pow(~)` only gets optimized when the power is an integer literal. That may explain the loss of performance. In theory that should mean that unwrapping the loop and still using `pow(~)` might perform equivalently, but I guess at that point we're not gaining much.

Thanks for benchmarking this!